dvol_python/dvol.py

```diff
@@ -166,9 +182,13 @@ def createBranch(self, volume, branch):
         self.output("Created branch %s/%s" % (volume, branch))
 
     def createVolume(self, name):
-        if self._directory.child(name).exists():
-            self.output("Error: volume %s already exists" % (name,))
-            raise VolumeAlreadyExists(name)
+        try:
+            if self._directory.child(name).exists():
```

Comment:
:question: 

I *think* this will:

* allow `.` as a valid name, but say that it already exists
* print a nice error for `..`, `../..`, `../././`, `../foo` etc
* allow `foo/../bar` and similar but maybe fail later or at least be *really* confusing to the user

I might be wrong, of course. That's why it would be good to write tests, or better, to have the computer write tests for you. 

Since you don't want to increase the scope to include this, and since the Python implementation might be disappearing anyway, I'd suggest adding a comment documenting some of the brokenness so we can fix this later.